src/client/render/render-core.js

```diff
@@ -29,6 +29,11 @@ let renderCore = function() {
 		// Make updates
 		renderCore.updateCallback(deltaTime);
 
+		// Update shaders
+		for (let shaderMaterial of renderCore.shaderMaterials) {
+			shaderMaterial.uniforms["time"].value += deltaTime;
```

Comment:
Uniforms are values that are shared on the GPU, meaning you'll only have to do this once for all shaders to be updated!
(You can use the skin loading bug to see this: If you spawn 50 fancy marbles at the same time, the shader is loaded + added to this list 50 times and progresses way too fast. Then if you add a different fancy marble, it matches the speed of the other shader.)

Also, it might be worth it to reset this to 0 between rounds, in case people leave their clients open a long time and start to get rounding errors. Don't know how fast that would happen, though.

Reply:
This makes a lot of sense with the issues I've been having. I had a bit of trouble getting the uniforms to work at all, so once it finally did the thing I was too scared to change it. Good to know!